src/classes/console/Console.js

```diff
@@ -0,0 +1,191 @@
+    * @property {string} magenta
+    * @property {string} cyan
+    * @property {string} gray
+    * @property {string} grey
+    * @property {string} brightgray
+    * @property {string} brightgrey
+    * @property {string} brightred
+    * @property {string} brightgreen
+    * @property {string} brightyellow
+    * @property {string} brightblue
+    * @property {string} brightmagenta
+    * @property {string} brightcyan
+    * @property {string} white
+    */
+
+  /**
+    * @memberof KomadaConsole
+    * @typedef {*} BackgroundColorTypes - All the valid background color types.
+    * @property {string} black
+    * @property {string} red
+    * @property {string} green
+    * @property {string} blue
+    * @property {string} magenta
+    * @property {string} cyan
+    * @property {string} gray
+    * @property {string} grey
+    * @property {string} brightgray
+    * @property {string} brightgrey
+    * @property {string} brightred
+    * @property {string} brightgreen
+    * @property {string} brightyellow
+    * @property {string} brightblue
+    * @property {string} brightmagenta
+    * @property {string} brightcyan
+    * @property {string} white
+    */
+
+  /**
+    * @memberof KomadaConsole
+    * @typedef {*} StyleTypes
+    * @property {string} normal
+    * @property {string} bold
+    * @property {string} dim
+    * @property {string} italic
+    * @property {string} underline
+    * @property {string} inverse
+    * @property {string} hidden
+    * @property {string} strikethrough
+    */
+
+  /**
+   * Logs everything to the console/writable stream.
+   * @param  {*} stuff The stuff we want to print.
+   * @param  {string} [type="log"] The type of log, particularly useful for coloring.
+   */
+  log(stuff, type = "log") {
+    stuff = KomadaConsole.flatten(stuff);
+    const message = this.colors ? this.colors[type.toLowerCase()].message : {};
+    const time = this.colors ? this.colors[type.toLowerCase()].time : {};
+    const timestamp = this.timestamps ? `[${moment().format("YYYY-MM-DD HH:mm:ss")}]` : null;
+    super.log(
+      stuff.split("\n").map(str => (timestamp ? `${this.timestamp(timestamp, time)} ${this.messages(str, message)}` : `${this.messages(str, message)}`)).join("\n"),
+    );
+  }
+
+  timestamp(timestamp, time) {
+    if (this.stdout !== process.stdout || this.stderr !== process.stderr) return timestamp;
+    return Colors.format(timestamp, time);
+  }
+
+  messages(string, message) {
+    if (!this.colors || this.stdout !== process.stdout || this.stderr !== process.stderr) return string;
+    return Colors.format(string, message);
+  }
+
+  /**
+   * Flattens our data into a readable string.
+   * @param  {*} data Some data to flatten
+   * @return {string}
+   */
+  static flatten(data) {
+    data = data.stack || data.message || data;
+    if (typeof data === "object" && typeof data !== "string" && !Array.isArray(data)) data = require("util").inspect(data, { depth: 0, colors: true });
```

Comment:
require("util") should be outside (no-inline-require) and the colors option should be editable.